web_poet/page_inputs.py

```diff
@@ -1,37 +1,35 @@
-import json
-from typing import Optional, Dict, List, TypeVar, Type
-
 import attrs
-from multidict import CIMultiDict
+import json
 import parsel
+from typing import Optional, Dict, Any
+
 from w3lib.encoding import (
     html_to_unicode,
     html_body_declared_encoding,
     resolve_encoding,
     http_content_type_encoding
 )
 
-from .utils import memoizemethod_noargs
+from web_poet._base import _HttpHeaders, _HttpBody
+from web_poet.utils import memoizemethod_noargs
 
-T_headers = TypeVar("T_headers", bound="HttpResponseHeaders")
 
-
-class HttpResponseBody(bytes):
+class HttpResponseBody(_HttpBody):
     """A container for holding the raw HTTP response body in bytes format."""
 
     def declared_encoding(self) -> Optional[str]:
         """ Return the encoding specified in meta tags in the html body,
         or ``None`` if no suitable encoding was found """
         return html_body_declared_encoding(self)
 
-    def json(self):
+    def json(self) -> Dict[Any, Any]:
```

Comment:
I think the result could be any JSON object (list, str, number), not necessarily a dict.

Reply:
You're right. Thanks for catching this! Reverted the mistake in https://github.com/scrapinghub/web-poet/pull/22/commits/c1f7675783cdb5ba970619f352f14417fe179655